### ISSUE TITLE
Introduce a keyword "board"

### ISSUE THREAD

**Issue by 0xc0170** [0xc0170/project_generator#54]
The board which would translate to MCU and CPU core and other details if required by tools.

This can be defined in the yaml file.
```
board:
    frdm-k64f
```

Or via an argument. There are already used arguments (b - build, p - project, -t tool). So what can be defined for board?)

```
project_generator -p blinky -br frdm-k64f -t make_gcc_arm
```

**Comment by 0xc0170** [comment]
This should also allow us to remove "core" which should be deduced from the board or mcu. Core is used for gcc-type tools , therefore it should be handled there that based on the mcu, it would get core.

**Comment by 0xc0170** [comment]
Fixed via #50
